From d20d99d3d9992d27f25b30916e0f85e50b4c9420 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:06:47 -0400
Subject: [powerpc] bpf: Fix DIVWU instruction opcode

Message-id: <1410545655-205645-179-git-send-email-dzickus@redhat.com>
Patchwork-id: 94185
O-Subject: [RHEL7 PATCH 178/626] powerpc/bpf: Fix DIVWU instruction opcode
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit a40a2b670706494610d794927b9aebe77e18af8d
Author: Vladimir Murzin <murzin.v@gmail.com>
Date:   Sat Sep 28 10:22:00 2013 +0200

    powerpc/bpf: Fix DIVWU instruction opcode

    Currently DIVWU stands for *signed* divw opcode:

    7d 2a 4b 96  divwu   r9,r10,r9
    7d 2a 4b d6  divw    r9,r10,r9

    Use the *unsigned* divw opcode for DIVWU.

    Suggested-by: Vassili Karpov <av1474@comtv.ru>
    Reviewed-by: Vassili Karpov <av1474@comtv.ru>
    Signed-off-by: Vladimir Murzin <murzin.v@gmail.com>
    Acked-by: Matt Evans <matt@ozlabs.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/include/asm/ppc-opcode.h b/arch/powerpc/include/asm/ppc-opcode.h
index 99f8790..3132bb9 100644
--- a/arch/powerpc/include/asm/ppc-opcode.h
+++ b/arch/powerpc/include/asm/ppc-opcode.h
@@ -222,7 +222,7 @@
 #define PPC_INST_MULLW   0x7c0001d6
 #define PPC_INST_MULHWU   0x7c000016
 #define PPC_INST_MULLI   0x1c000000
-#define PPC_INST_DIVWU   0x7c0003d6
+#define PPC_INST_DIVWU   0x7c000396
 #define PPC_INST_RLWINM   0x54000000
 #define PPC_INST_RLDICR   0x78000004
 #define PPC_INST_SLW   0x7c000030
-- 
1.7.1